checker/src/org/checkerframework/checker/units/UnitsRelationsTools.java

```diff
@@ -178,17 +207,42 @@ private static boolean hasNoPrefix(/*@Nullable*/ final AnnotationValue annotatio
     }
 
     /**
-     * Removes the Prefix value from an Annotation, by constructing and returning a copy of its base SI unit's Annotation
-     * @param elements
-     *            Element Utilities from a checker's processing environment,
-     *            typically obtained by calling env.getElementUtils() in init()
-     *            of a Units Relations implementation
-     * @param unitsAnnotation An AnnotationMirror representing a Units Annotation
-     * @return The base SI Unit's AnnotationMirror, or null if the base SI Unit cannot
-     *         be constructed
+     * Given an Annotated Type, this method returns the Annotation representing
+     * the unit that this Annotated Type has.
+     *
+     * @param annoType an Annotated Type
+     * @return an AnnotationMirror representing the unit that this Annotated
+     *         Type has
+     */
+    public static /*@Nullable*/ AnnotationMirror getUnit(/*@Nullable*/ final AnnotatedTypeMirror annoType) {
+        if (annoType == null) {
+            return null;
+        }
+        // return the first unit (should always be the only unit) in the
```

Comment:
For the time being, the intended usage of Units Checker would only use 1 annotation per slot. For something like Joules which could be represented as `@N @m` we would provide the `@J` annotation which would be encoded as the result of multiplying `@N` with `@m`, likewise for division rules.

I can see the use of allowing multiple units annotations per slot though, and that would simplify a few other annotations we already have, eg by breaking up `@mPERs` into `@m` and `@PER(@s)`, allowing us to express units that are defined as a combination of base SI units more easily without having to create many xPERy combinations as separate and specific annotations.

Should I change the checker to support this latter design?